tests-beta/android/MASVS-CRYPTO/MASTG-TEST-0208.md

```diff
@@ -0,0 +1,23 @@
+---
+platform: android
+title: Inappropriate Key Sizes 
+id: MASTG-TEST-0208
+type: [static]
+weakness: MASWE-0009
+---
+
+## Overview
+
+The application appears to utilize a symmetric algorithm  with a weak key configuration, making it susceptible to multiple vulnerabilities and potential security breaches.
```

Comment:
```suggestion
In this test case, we will look for the use inappropriate key sizes in Android apps. To do this, we need to focus on the cryptographic frameworks and libraries that are available in Android and the methods that are used to generate cryptographic keys.

In Android the class `KeyGenerator` offers different [encryption algorithms](https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#KeyGenerator) for key creation. 

There are a variety of different [KeyGenerator.init()](https://developer.android.com/reference/javax/crypto/KeyGenerator#public-methods) methods to initialise the key generation, where some require random bytes. The entropy of these random bytes will rely on the  `SecureRandom` implementation in the available security provider, otherwise a system-provided source of randomness will be used.



```